integrasjonspunkt/src/test/java/no/difi/meldingsutveksling/noarkexchange/altinn/IntegrasjonspunktReceiveImplIntegrationTest.java

```diff
@@ -0,0 +1,75 @@
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.boot.test.WebIntegrationTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBElement;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+import javax.xml.transform.stream.StreamSource;
+import java.io.IOException;
+
+import static org.mockito.Mockito.*;
+
+/**
+ * Integration test for {@link IntegrajonspunktReceiveImpl} (via {@link InternalQueue}).
+ *
+ * Mock overrides are configured in {@link no.difi.meldingsutveksling.noarkexchange.IntegrasjonspunktIntegrationTestConfig}
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(IntegrasjonspunktApplication.class)
+@WebIntegrationTest
+@ActiveProfiles("test")
+public class IntegrasjonspunktReceiveImplIntegrationTest {
+
+    @Autowired
+    InternalQueue internalQueue;
+
+    @Autowired
+    IntegrajonspunktReceiveImpl integrajonspunktReceive;
+
+    @Test
+    public void receiveMessageTest() throws LookupException, JAXBException, IOException, MessageException {
+        AppReceiptType appReceiptTypeMock = mock(AppReceiptType.class);
+        when(appReceiptTypeMock.getType()).thenReturn("OK");
+        PutMessageResponseType putMessageResponseTypeMock = mock(PutMessageResponseType.class);
+        when(putMessageResponseTypeMock.getResult()).thenReturn(appReceiptTypeMock);
+        NoarkClient noarkClientMock = mock(NoarkClient.class);
+        when(noarkClientMock.sendEduMelding(any(PutMessageRequestType.class))).thenReturn(putMessageResponseTypeMock);
+        integrajonspunktReceive.setLocalNoark(noarkClientMock);
+
+        IntegrajonspunktReceiveImpl integrajonspunktReceiveSpy = spy(integrajonspunktReceive);
+        doReturn("42".getBytes()).when(integrajonspunktReceiveSpy).decrypt(any(Payload.class));
+        PutMessageRequestType putMessageRequestTypeMock = mock(PutMessageRequestType.class);
+        doReturn(putMessageRequestTypeMock).when(integrajonspunktReceiveSpy).convertAsicEntrytoEduDocument(any(byte[].class));
+        doNothing().when(integrajonspunktReceiveSpy).sendReceiptOpen(any(StandardBusinessDocumentWrapper.class));
+        internalQueue.setIntegrajonspunktReceiveImpl(integrajonspunktReceiveSpy);
+
+        JAXBElement<StandardBusinessDocument> fromDocument;
+        JAXBContext ctx = JAXBContext.newInstance(no.difi.meldingsutveksling.noarkexchange.schema.receive.StandardBusinessDocument.class);
+        Unmarshaller unmarshaller = ctx.createUnmarshaller();
+        fromDocument = unmarshaller.unmarshal(new StreamSource(getClass().getClassLoader().getResourceAsStream("1466595652965.xml")), no.difi.meldingsutveksling.noarkexchange.schema.receive.StandardBusinessDocument.class);
```

Comment:
Kan denne byttes ut med kall til TestData evt. refaktorert og/eller lagt til metode for akkurat denne. Regner vi kan komme til å måtte lese inn sånne xml filer i andre tester.